.github/workflows/ci.yaml

```diff
@@ -782,3 +698,158 @@ jobs:
           keep-tags-regexes: ^pr
           prune-tags-regexes: ^main-
           prune-untagged: true
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: coder
+          path: |
+            ./build/*.zip
+            ./build/*.tar.gz
+            ./build/*.deb
+          retention-days: 7
+
+  deploy:
+    name: "deploy"
+    runs-on: ${{ github.repository_owner == 'coder' && 'buildjet-16vcpu-ubuntu-2204' || 'ubuntu-latest' }}
+    timeout-minutes: 30
+    needs:
+      - changes
+      - build
+    if: |
+      github.ref == 'refs/heads/main' && !github.event.pull_request.head.repo.fork
+      && needs.changes.outputs.docs-only == 'false'
```

Comment:
```suggestion
```
Similarly, when we have `needs: build`, we do not need this check. as this job will be skipped if `build` is skipped.

Reply:
I think the check should be duplicated.